### PR TITLE
[#189] Update CPR to 1.4.0 and remove extra libs

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses #189

CPR's internal version number was not bumped for 1.4.0, so 1.3.0 is still referenced in many places in the final package

### CODE REVIEW

**Comment by trel** [comment]
oh, that's... silly.

**Comment by trel** [comment]
Did you say cpr isn't packaged by the OSs, in general?   So we can go bigger than 1.4.0 with little extra effort?   They're up to 1.10.x now...

**Comment by SwooshyCueb** [comment]
Not if we still want to support centos 7 and Ubuntu 18.04. Their curl isn't new enough.

**Comment by trel** [comment]
Oh.  Blasted.

File: versions.json [hunk 1]
```diff
@@ -136,14 +136,15 @@
         "fpm_directories": ["include","lib"]
     },
     "cpr": {
-        "commitish": "1.3.0",
-        "version_string": "1.3.0",
+        "git_repository": "https://github.com/libcpr/cpr",
```
Comment (korydraughn) [inline versions.json:139]: Once you're happy with it, we'll get @trel or @alanking to update our fork. Then we can remove the ref to their git repo.

Comment (SwooshyCueb) [reply versions.json:139]: Since we're not patching, I don't see a reason we shouldn't use the upstream repo

Comment (korydraughn) [reply versions.json:139]: That's correct. We could use the upstream repo, but I believe we maintain forks for all externals to protect us from issues with upstream.

@trel would know the exact reason for why we do that.

Always open to hearing how to make this better.

**Comment by trel** [comment]
yes, we keep forks in case of shenanigans upstream.    we can easily re-sync our forks.

**Comment by trel** [comment]
our cpr is now synced with upstream.

**Comment by korydraughn** [comment]
Is this work in service of libstdc++ or iRODS 4.3.1, 4.4, or 5.0?

**Comment by SwooshyCueb** [comment]
> our cpr is now synced with upstream.

We still need the tags

**Comment by trel** [comment]
ooh, try again.

**Comment by SwooshyCueb** [comment]
I think I'm happy with this. Any objections before I \#?

**Comment by korydraughn** [comment]
Does anything for 4.3.1 need this?

**Comment by SwooshyCueb** [comment]
IIRC the extra curl library causes problems for one of the plugins, so maybe?

**Comment by korydraughn** [comment]
Hmm.

I think the indexing plugin is the only thing that relies on cpr, but please check to see if there's any other projects that depend on it.

**Comment by trel** [comment]
looks good if we determine that 1.4.0 is sufficient.

**Comment by korydraughn** [comment]
I agree.

**Comment by SwooshyCueb** [comment]
\#'d.

Once this is in the package repos I'll submit PRs updating irods and the indexing plugin

**Comment by SwooshyCueb** [comment]
What else needs to happen before this gets merged?

**Comment by korydraughn** [comment]
Nothing AFAIK. It's a matter of @alanking @trel clicking the merge button.

Have you confirmed which projects are affected by this change?
Which projects are those?

**Comment by SwooshyCueb** [comment]
I thought the server used it as well, but it looks like it's just the indexing plugin.

**Comment by korydraughn** [comment]
Very happy that only a single project depends on it. Keeps things simple.

**Pull request merged**